src/common/components/play/PreGameModal.js

```diff
@@ -89,14 +93,26 @@ export default class PreGameModal extends React.Component {
     this.setState({ selectedDeckIdx });
   }
 
+  handleSetPassword = (e) => {
+    this.setState({ enteredPassword: e.target.value });
+  }
+
   handleStartGame = () => {
-    this.props.onStartGame(this.format.name, this.deckForGame);
+    const { options, onStartGame } = this.props;
+    const { enteredPassword } = this.state;
+
+    if (options.passwordToJoin && enteredPassword !== options.passwordToJoin) {
+      alert('Incorrect password!');
```

Comment:
Should this be displayed in the modal rather than as an alert?

Reply:
I can't think of a good way to do that that's not gross :-/